ciao-controller/internal/datastore/datastore.go

```diff
@@ -2280,3 +2257,61 @@ func (ds *Datastore) UnMapExternalIP(address string) error {
 
 	return nil
 }
+
+func (ds *Datastore) generateCNCIWorkload() {
+	// generate the CNCI workload.
+	config := `---
+#cloud-config
+users:
+  - name: cloud-admin
+    gecos: CIAO Cloud Admin
+    lock-passwd: false
+    passwd: $6$rounds=4096$w9I3hR4g/hu$AnYjaC2DfznbPSG3vxsgtgAS4mJwWBkcR74Y/KHNB5OsfAlA4gpU5j6CHWMOkkt9j.9d7OYJXJ4icXHzKXTAO.
+    sudo: ALL=(ALL) NOPASSWD:ALL
+...
+`
+	cpus := payloads.RequestedResource{
+		Type:      payloads.VCPUs,
+		Value:     4,
+		Mandatory: false,
+	}
+
+	mem := payloads.RequestedResource{
+		Type:      payloads.MemMB,
+		Value:     128,
```

Comment:
Currently, we have two separate workload_resources.csv files, one for SingleVM and one for normal clusters.  The singleVM version of workload_resources.csv gives the CNCI 128MB ( we want to keep it small  for SingleVM ) but the controller copy of the file, the one used in real clusters, gives it 2048MB disk and 2048MB RAM.  Is this commit using 128MB for all CNCI instances regardless of whether they are deployed on SingleVM or a real cluster?  If so I don't think it's correct.  We'll probably run into memory issues on real deployments.  I think we need to be able to specify different CNCI workloads.

@mcastelino Can you comment on this?